src/ai/backend/manager/models/audit_logs.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Optional, Sequence
+
+import graphene
+import sqlalchemy as sa
+from dateutil.parser import parse as dtparse
+from graphene.types.datetime import DateTime as GQLDateTime
+from sqlalchemy.dialects import postgresql as pgsql
+from sqlalchemy.engine.row import Row
+
+from ai.backend.common.logging import BraceStyleAdapter
+
+from .base import Item, PaginatedList, metadata, simple_db_mutate
+from .minilang.ordering import QueryOrderParser
+from .minilang.queryfilter import QueryFilterParser
+from .user import UserRole
+
+if TYPE_CHECKING:
+    from .gql import GraphQueryContext
+log = BraceStyleAdapter(logging.getLogger(__name__))
+__all__: Sequence[str] = (
+    "audit_logs",
+    "AuditLog",
+    "AuditLogList",
+    "AuditLogInput",
+    "CreateAuditLog",
+)
+
+audit_logs = sa.Table(
+    "audit_logs",
+    metadata,
+    sa.Column("user_id", sa.String(length=256), index=True),
+    sa.Column("access_key", sa.String(length=20), index=True),
+    sa.Column("email", sa.String(length=64), index=True),
+    sa.Column(
+        "action",
+        sa.Enum("CREATE", "CHANGE", "DELETE", name="auditlogs_action", create_type=False),
+        index=True,
+    ),
+    sa.Column("data", pgsql.JSONB()),
+    sa.Column(
+        "target_type",
+        sa.Enum("user", "keypairs", "vfolder", name="auditlogs_targettype", create_type=False),
+        index=True,
+    ),
+    sa.Column("target", sa.String(length=64), index=True),
+    sa.Column("created_at", sa.DateTime(timezone=True), server_default=sa.func.now(), index=True),
+)
+
+
+class AuditLog(graphene.ObjectType):
```

Comment:
There should be a security consideration to prevent unprivileged users from accessing audit log of others. Let's refactor this object to raise error if the requester is non-admin but variables referencing each user (`user_id`, `access_key` and `email`) does not match requester's identity.